forc-plugins/forc-client/src/util/tx.rs

```diff
@@ -81,6 +180,107 @@ async fn collect_account_balances(
+                anyhow::bail!("Your wallet does not have any funds to pay for the transaction.\
+                                      \n\nIf you are interacting with a testnet consider using the faucet.\
+                                      \n-> {target} network faucet: {faucet_link}\
+                                      \nIf you are interacting with a local node, consider providing a chainConfig which funds your account.")
+            }
+            print_account_balances(&accounts, &account_balances);
+
+            let mut account_index;
+            loop {
+                print!("\nPlease provide the index of account to use for signing: ");
+                std::io::stdout().flush()?;
+                let mut input_account_index = String::new();
+                std::io::stdin().read_line(&mut input_account_index)?;
+                account_index = input_account_index.trim().parse::<usize>()?;
+                if accounts.contains_key(&account_index) {
+                    break;
+                }
+                let options: Vec<String> = accounts.keys().map(|key| key.to_string()).collect();
+                println_warning(&format!(
+                    "\"{}\" is not a valid account.\nPlease choose a valid option from {}",
+                    account_index,
+                    options.join(","),
+                ));
+            }
+
+            let secret_key = secret_key_from_forc_wallet(&wallet_path, account_index, password)?;
+
+            let bech32 = bech32_from_secret(&secret_key)?;
+            // TODO: Do this via forc-wallet once the functionality is exposed.
+            let question = format!(
+                "Do you agree to sign this transaction with {}? [y/N]: ",
+                bech32
+            );
+            let accepted = ask_user_yes_no_question(&question)?;
+            if !accepted {
+                anyhow::bail!("User refused to sign");
+            }
+
+            Some(secret_key)
+        }
+        WalletSelectionMode::Manual => select_manual_secret_key(default_sign, signing_key),
+    };
+    Ok(signing_key)
+}
+
+pub(crate) async fn update_proxy_contract_target(
+    provider: Provider,
+    secret_key: SecretKey,
+    proxy_contract_id: ContractId,
+    new_target: ContractId,
+) -> Result<()> {
+    abigen!(Contract(
+        name = "ProxyContract",
+        abi = "forc-plugins/forc-client/src/util/proxy_contract-abi.json"
+    ));
+
+    let wallet = WalletUnlocked::new_from_private_key(secret_key, Some(provider));
+
+    let proxy_contract = ProxyContract::new(proxy_contract_id, wallet);
+
+    // TODO: what happens if the call fails? Does 'FuelCallResponse' is returned as Err() in that case?
+    proxy_contract
+        .methods()
+        .set_proxy_target(new_target)
+        .call()
+        .await?;
+    println!(
+        "  0x{} proxy contract's target to {}",
+        "Updated".green(),
+        new_target
+    );
+    Ok(())
+}
```

Comment:
Is this just refactoring the code that was previously in `finalize_signed`? Or is the behavior changing?

It would be great to add unit tests for these functions

Reply:
This is basically moving the inside of finalize signed as a separate function. I will try to add tests for this.